src/controllers/playback/video/index.js

```diff
@@ -1504,6 +1505,18 @@ import { setBackdropTransparency, TRANSPARENCY_LEVEL } from '../../../components
             playbackManager.toggleFullscreen(currentPlayer);
         });
 
+        // Set currently playing item for favorite button
+        const playingItem = playbackManager.getPlayerState(currentPlayer).NowPlayingItem;
+        let btnUserRating = view.querySelector('.btnUserRating');
+
+        if (itemHelper.canRate(playingItem)) {
+            btnUserRating.classList.remove('hide');
+            btnUserRating.setItem(playingItem);
+        } else {
+            btnUserRating.classList.add('hide');
+            btnUserRating.setItem(null);
+        }
+
```

Comment:
You should do this in `updateDisplayItem`.
When playing a playlist, the rating button will always mark the first item.